sqlmesh/core/plan/definition.py

```diff
@@ -744,8 +748,8 @@ def _refresh_dag_and_ignored_snapshots(self) -> None:
                 ).sorted
             }
 
-        self._add_restatements()
         self.__missing_intervals = None
+        self._add_restatements()
```

Comment:
@eakmanrq another potential issue. We used to add restatements before resetting previous intervals, which means that the `self.start` used to compute restatement intervals could've been derived from previous missing intervals.